RepoCleanup/Functions/MigrateXsdSchemasFunction.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+using LibGit2Sharp;
+
+using RepoCleanup.Application.Commands;
+using RepoCleanup.Application.CommandHandlers;
+using RepoCleanup.Models;
+using RepoCleanup.Services;
+using RepoCleanup.Utils;
+
+namespace RepoCleanup.Functions
+{
+    public static class MigrateXsdSchemasFunction
```

Comment:
If I where to comment on this as if it was a controller (which you could compare it to) I would say there is too much going on here that should be in the CommandHandler (creating repositores, orchestrating the flow), The function should be about the UI part, gathering and validating required input in order to fire the right commands/commandHandler. Then I would consider creating one command/commandHandler for the entire operation. This will probably force us to separate out some logic if we need it somewhere else, but in the CQRS pattern we're also missing a domain layer. The handler is about orchestrating the flow.

Reply:
Created a new command that replaces the other attempts. Thanks for the guidance.